src/plugin_system/testing/harness.py

```
@@ -0,0 +1,314 @@
+    types: List[type] = [
+        ConnectionError, TimeoutError,        # builtins
+        socket.gaierror, socket.timeout,      # DNS / socket timeouts
+        ssl.SSLError,
+        urllib.error.URLError,
+        http.client.HTTPException,
+    ]
+    try:  # requests is optional; cover its whole error tree when present
+        import requests
+        types.append(requests.exceptions.RequestException)
+    except Exception:  # pragma: no cover - requests not installed
+        pass
```

Ruff (0.15.15)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)